### PR TITLE
Add compilers configuration

### PR DESCRIPTION
**Pull request by Aertalon**
Add tpl file with common/gcc/clang warning flags

### CODE REVIEW

**Comment by oliverlee** [comment]
`defs.bzl.tpl` is an input file. You need to add these to actually configure it:
https://github.com/oliverlee/turtle/blob/master/configure.bzl
https://github.com/oliverlee/turtle/blob/master/WORKSPACE#L12

Then you actually want to make use of the options when building targets:
https://github.com/oliverlee/turtle/blob/master/BUILD#L1
https://github.com/oliverlee/turtle/blob/master/BUILD#L36

It may be worth writing wrappers for `cc_binary`,  `cc_library`,  `cc_test` so that it automatically applies the options.

**Comment by Aertalon** [comment]
I'll take a look at how to define the wrappers

**Comment by oliverlee** [comment]
FYI this is not "the right way" but I haven't invested enough time to figure out how to do this properly in Bazel.

**Comment by Aertalon** [comment]
I should probably squash everything and restart from zero..

**Comment by oliverlee** [comment]
Up to you. This is your repo :)

File: BUILD [hunk 1]
```diff
@@ -1,6 +1,6 @@
-exports_files(
-    [".clang-tidy"],
-    visibility = ["//:__subpackages__"],
+load(
+    "@local_config//:defs.bzl",
+    "PROJECT_DEFAULT_COPTS",
```
Comment (oliverlee) [inline BUILD:3]: Have you tried using this for any targets?

**Comment by Aertalon** [comment]
Replaced by squashed version

**Pull request closed without merging**